### PR TITLE
sdk/javascript: initial implementation of crypto + models, passing all test vectors

### PR DESCRIPTION
**Pull request by Jaguar0625**

### CODE REVIEW

File: sdk/javascript/src/CryptoTypes.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	/**
+	 * Creates a zeroed hash.
+	 * @returns {Hash256} Zeroed hash.
+	 */
+	static zero() {
+		return new Hash256(new Uint8Array(Hash256.SIZE));
+	}
+}
+
+Hash256.SIZE = 32;
```
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply sdk/javascript/src/CryptoTypes.js:25]: they are equivalent, but eslint doesn't yet support static fields; so given
```js
	static SIZE = 32;
```

eslint fails with
```
Parsing error: Unexpected token =
```

File: sdk/javascript/src/symbol/idGenerator.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	return result;
+};
+
+/**
+ * Generates a mosaic id from an owner address and a nonce.
+ * @param {Address} ownerAddress Owner address.
+ * @param {number} nonce Nonce.
+ * @returns {BigInt} Computed mosaic id.
+ */
+const generateMosaicId = (ownerAddress, nonce) => {
```
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply sdk/javascript/src/symbol/idGenerator.js:26]: we treat it as unsigned, so no

File: sdk/javascript/src/Network.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+ * Provides utility functions for finding a network.
+ */
+const NetworkLocator = {
+	/**
+	 * Finds a network with a specified name within a list of networks.
+	 * @param {array<Network>} networks List of networks to search.
+	 * @param {array<string>|string} singleOrMultipleNames Names for which to search.
+	 * @returns {Network} First network with a name in the supplied list.
+	 */
+	findByName: (networks, singleOrMultipleNames) => {
```
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply sdk/javascript/src/Network.js:84]: added multiple support to match with python sdk, so would remove from both if at all. 
originally it was needed when NEM and Symbol networks didn't share network names (i.e. testnet vs public-test)

File: sdk/javascript/test/symbol/idGenerator_spec.js [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+	improperQualified: ['.', '..', '...', '.a', 'b.', 'a..b', '.a.b', 'b.a.']
+};
+
+describe('idGenerator', () => {
+	// region generateMosaicId
+
+	describe('generateMosaicId', () => {
+		it('generates correct id', () => {
+			// Act:
+			const mosaicId = generateMosaicId(new Address('TATNE7Q5BITMUTRRN6IB4I7FLSDRDWZA37JGO5Q'), 812613930);
```
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply sdk/javascript/test/symbol/idGenerator_spec.js:20]: probably, but hopefully vectors cover this :😅

**Review by gimre-xymcity — APPROVED** [review]

**Pull request merged**